src/cursor.rs

```diff
@@ -469,9 +469,12 @@ impl Cursor {
     /// # Return value
     ///
     /// Returns a vector containing the BSON documents that were read.
-    pub fn next_batch(&mut self) -> Result<Vec<bson::Document>> {
-        let n = self.batch_size;
-        self.next_n(n)
+    pub fn get_current_batch(&mut self) -> Result<Vec<bson::Document>> {
```

Comment:
Given that the default batch size is being changed zero, the implementation of this method needs to be changed as well. Since a batch size of 0 means "let the server decide the batch size", I don't think it makes sense to hard code a specific number of documents to fetch in this method, so the most straightforward implementation would be to simply return the remaining documents from the current batch. While this changes what the method will return in the case that the buffer is partially empty, I think this is more in line with my expectations of what this method should be doing.

The name `next_batch` doesn't quite seem to fit with the proposed new behavior of the method, so I also propose changing it to `get_current_batch`, which is a more accurate description.